site/viamguides/src/yolo-training/yolo-training.md

```diff
@@ -0,0 +1,307 @@
+   ![model details](assets/model-details.png)
+
+
+Well done, you've just created your own custom YOLO model tailored to your specific objects! Now you can deploy it to your machine.
+
+<!-- ------------------------ -->
+## Configure the YOLO-ONNX ML Model
+Duration: 5
+
+Now that you have a trained model, let's add it to your Viam machine using the `yolo-onnx` ML model service.
+
+1. In [the Viam app](https://app.viam.com/fleet/locations), find the **CONFIGURE** tab.
+1. Click the **+** icon in the left-hand menu and select **Service**.
+1. Select `ML Model`, then search for and select the `hipsterbrown:mlmodel:yolo-onnx` module from the registry. Leave the default name `mlmodel-1` for now, then click **Create**.
+   <img src="assets/add-yolo-onnx.png" alt="add yolo-onnx module" width="450" />
+1. Notice adding this module adds the ML Model service called `mlmodel-1` and the `yolo-onnx` module from the Viam registry. You'll see configurable cards on the right and the corresponding parts listed in the left sidebar.
+   ![ML model service added](assets/mlmodel-added.png)
+1. Click on the JSON tab in the left sidebar above your machine name to display the full configuration JSON for your machine.
+   ![select JSON view](assets/switch-json-view.png)
+   ![overview of JSON config](assets/json-config-overview.png)
+1. At the bottom of the configuration, add a `"packages"` field with the details of your YOLO model from the registry:
+   ```json
+    "packages": [{
+      "name": "my-yolo-detector",
+      "package": "<organization-id>/my-yolo-detector",
+      "type": "ml_model",
+      "version": "<version>"
+    }]
+   ```
+   The name should match what you see displayed on the model page in the Registry.
+   Replace the "\<organization-id>" with your Organization ID, which can be found in **Settings and invites** page. If you change your model to be public in the Registry, you can use your public namespace instead.
+   <img src="assets/view-org-settings.png" alt="view org settings" width="450" />
+   <img src="assets/view-org-details.png" alt="view org details" width="450" />
+   Replace the "\<version>" with the latest version displayed in the Registry, i.e. `2025-04-16T17-29-48`.
+   <img src="assets/configure-packages.png" alt="configure mlmodel package" width="450" />
+
+1. Switch back to the Builder tab to update the JSON configuration of the `mlmodel-1` service. You need to provide the paths to your model and labels files. You can get those by referencing your trained model package from the Registry.
+   ```json
+   {
+     "package_reference": "<organization-id>/my-yolo-detector",
+     "model_path": "${packages.ml_model.my-yolo-detector}/best.onnx",
+     "label_path": "${packages.ml_model.my-yolo-detector}/labels.txt"
+   }
+   ```
+   <img src="assets/configure-yolo-onnx.png" alt="configure yolo-onnx attributes" width="450" />
+   The `"package_reference"` should match the `"package"` field you set in the previous step.
```

Comment:
This is a temporary workaround for a bug in the app that prevents all ML models from being displayed in the Registry picker UI for custom ML model services. A fix was just merged and should go out in the next app release. I will update this codelab with those steps once that is live.